homeassistant/components/isy994/services.py

```diff
@@ -0,0 +1,408 @@
+    "stop",
+    "enable",
+    "disable",
+    "enable_run_at_startup",
+    "disable_run_at_startup",
+]
+
+
+def valid_isy_commands(value: Any) -> str:
+    """Validate the command is valid."""
+    value = str(value).upper()
+    if value in COMMAND_FRIENDLY_NAME.keys():
+        return value
+    raise vol.Invalid("Invalid ISY Command.")
+
+
+SCHEMA_GROUP = "name-address"
+
+SERVICE_SYSTEM_QUERY_SCHEMA = vol.Schema(
+    {vol.Optional(CONF_ADDRESS): cv.string, vol.Optional(CONF_ISY): cv.string}
+)
+
+SERVICE_SET_RAMP_RATE_SCHEMA = {
+    vol.Required(CONF_VALUE): vol.All(vol.Coerce(int), vol.Range(0, 31))
+}
+
+SERVICE_SET_VALUE_SCHEMA = {
+    vol.Required(CONF_VALUE): vol.All(vol.Coerce(int), vol.Range(0, 255))
+}
+
+SERVICE_SEND_RAW_NODE_COMMAND_SCHEMA = {
+    vol.Required(CONF_COMMAND): vol.All(cv.string, valid_isy_commands),
+    vol.Optional(CONF_VALUE): vol.All(vol.Coerce(int), vol.Range(0, 255)),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT): vol.All(vol.Coerce(int), vol.Range(0, 120)),
+    vol.Optional(CONF_PARAMETERS, default={}): {cv.string: cv.string},
+}
+
+SERVICE_SEND_NODE_COMMAND_SCHEMA = {
+    vol.Required(CONF_COMMAND): vol.In(VALID_NODE_COMMANDS)
+}
+
+SERVICE_SET_VARIABLE_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_ADDRESS, CONF_TYPE, CONF_NAME),
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_NAME, SCHEMA_GROUP): cv.string,
+            vol.Inclusive(CONF_ADDRESS, SCHEMA_GROUP): vol.Coerce(int),
+            vol.Inclusive(CONF_TYPE, SCHEMA_GROUP): vol.All(
+                vol.Coerce(int), vol.Range(1, 2)
+            ),
+            vol.Optional(CONF_INIT, default=False): bool,
+            vol.Required(CONF_VALUE): vol.Coerce(int),
+            vol.Optional(CONF_ISY): cv.string,
+        }
+    ),
+)
+
+SERVICE_SEND_PROGRAM_COMMAND_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_ADDRESS, CONF_NAME),
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_NAME, SCHEMA_GROUP): cv.string,
+            vol.Exclusive(CONF_ADDRESS, SCHEMA_GROUP): cv.string,
+            vol.Required(CONF_COMMAND): vol.In(VALID_PROGRAM_COMMANDS),
+            vol.Optional(CONF_ISY): cv.string,
+        }
+    ),
+)
+
+SERVICE_RUN_NETWORK_RESOURCE_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_ADDRESS, CONF_NAME),
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_NAME, SCHEMA_GROUP): cv.string,
+            vol.Exclusive(CONF_ADDRESS, SCHEMA_GROUP): vol.Coerce(int),
+            vol.Optional(CONF_ISY): cv.string,
+        }
+    ),
+)
+
+
+@callback
+def async_setup_services(hass: HomeAssistantType):
+    """Create and register services for the ISY integration."""
+    existing_services = hass.services.async_services().get(DOMAIN)
+    if existing_services and any(
+        service in INTEGRATION_SERVICES for service in existing_services.keys()
+    ):
+        # Integration-level services have already been added. Return.
+        return
+
+    async def async_system_query_service_handler(service):
```

Comment:
@shbatm  Please disregard my earlier feedback on multiple context switches.  Since you are providing the node address it should only drop one job in the executor.